sentinel/tests/unit/schedule/index.js

```diff
@@ -324,4 +329,66 @@ describe('scheduler', () => {
     });
   });
 
+  // Regression test to prevent again jobs launched 1 sec before time
+  // to NOT being skipped (https://github.com/medic/cht-core/issues/6634)
+  describe('Scheduling without gaps', () => {
+
+    let purgeMoment;
+
+    beforeEach(() => {
+      realSetTimeout = setTimeout;
+      // default value in the past so if purge is never called because an
+      // unexpected error the math used on this variable will fail
+      purgeMoment = moment().subtract(10, 'seconds');
+      unit = rewire('../../../src/schedule/index'); // rewire AFTER we fake time setTimeout
+      sinon.stub(transitionsLib.dueTasks, 'execute').resolves();
+      sinon.stub(reminders, 'execute').resolves();
+      sinon.stub(replications, 'execute').resolves();
+      sinon.stub(outbound, 'execute').resolves();
+      sinon.stub(backgroundCleanup, 'execute').resolves();
+
+      sinon.stub(purgeLib, 'purge').callsFake(() => {
+        purgeMoment = moment();
+        logger.debug('Purge time: %s', purgeMoment);
+      });
+    });
+
+    afterEach(() => {
+      clearInterval(unit.__get__('interval'));
+      sinon.restore();
+    });
+
+    it('should not skip job scheduled less than 1 sec before time set and launch it 1 sec later', () => {
+      sinon.stub(scheduling, 'getSchedule')
+        .returns(later.parse.cron('* * * * * *', true));  // each 1 second
+      const now = moment();
+      unit.init();
+      return nextTick(1100)
+        .then(() => {
+          expect(purgeMoment.seconds()).to.equal(now.seconds() + 1);
+        });
+    }).slow(1200);
+
+    it('should not skip job scheduled milliseconds ahead the time set and launch it 1 sec later', () => {
+      sinon.stub(scheduling, 'getSchedule')
+        .returns(later.parse.cron('*/2 * * * * *', true));  // each 2 seconds, even seconds
+      let wait = 0;
+      const now = moment();
+      if (now.seconds() % 2 === 1) {
```

Comment:
Doesn't this mean current time already has odd seconds and we're actually waiting for an even second (contrary to your comment).

Reply:
Yes, that is the intention of that `if` block, I'll add more comments inline to make it more clear.